src/zuuid.c

```diff
@@ -123,6 +136,9 @@ zuuid_set_str (zuuid_t *self, const char *source)
             return -1;
         self->uuid [byte_nbr] = (byte) value;
     }
+    
+    self->str[ZUUID_STR_LEN-1] = '\0';
```

Comment:
OK, fixed it: https://github.com/zeromq/czmq/pull/872

On Thu, Jan 1, 2015 at 6:16 PM, Asmod4n <notifications@github.com> wrote:

> In src/zuuid.c
> <https://github.com/zeromq/czmq/pull/871#discussion-diff-22400650>:
>
> > @@ -123,6 +136,9 @@ zuuid_set_str (zuuid_t *self, const char *source)
> >              return -1;
> >          self->uuid [byte_nbr] = (byte) value;
> >      }
> > +
> > +    self->str[ZUUID_STR_LEN-1] = '\0';
>
> Doesn't this have to be
>
> self->str[ZUUID_STR_LEN] = '\0';
>
> ?
>
> @mosconi <https://github.com/mosconi> @hintjens
> <https://github.com/hintjens>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/czmq/pull/871/files#r22400650>.
>